runtime/src/snapshot_utils/snapshot_storage_rebuilder.rs

```diff
@@ -0,0 +1,363 @@
+//! Provides interfaces for rebuilding snapshot storages
+
+use {
+    crate::{
+        accounts_db::{AccountStorageEntry, AppendVecId, AtomicAppendVecId, SlotStores},
+        serde_snapshot::{
+            self, remap_and_reconstruct_single_storage, snapshot_storage_lengths_from_fields,
+            SerdeStyle,
+        },
+    },
+    crossbeam_channel::{select, unbounded, Receiver, Sender},
+    dashmap::DashMap,
+    log::info,
+    rayon::{
+        iter::{IntoParallelIterator, ParallelIterator},
+        ThreadPool, ThreadPoolBuilder,
+    },
+    solana_sdk::clock::Slot,
+    std::{
+        collections::HashMap,
+        fs::File,
+        io::BufReader,
+        path::PathBuf,
+        sync::{
+            atomic::{AtomicUsize, Ordering},
+            Arc, Mutex,
+        },
+        time::Instant,
+    },
+};
+
+/// Stores state for rebuilding snapshot storages
+#[derive(Debug)]
+pub struct SnapshotStorageRebuilder {
+    /// Receiver for unpacked snapshot storage files
+    file_receiver: Receiver<PathBuf>,
+    /// Number of threads to rebuild with
+    num_threads: usize,
+    /// Snapshot storage lengths - from the snapshot file
+    snapshot_storage_lengths: HashMap<Slot, HashMap<usize, usize>>,
```

Comment:
what is usize and usize? appendvecid and len? I think we can maybe typedef this or use appendvecid for the key?

Reply:
This should be `SerializedAppendVecId` and length. I'll make that type pub so I can access it here.